v2/clients/http/deviceprofile.go

```diff
@@ -44,3 +49,87 @@ func (client *DeviceProfileClient) Update(ctx context.Context, reqs []requests.D
+	}
+	return res, nil
+}
+
+func (client *DeviceProfileClient) AllDeviceProfiles(ctx context.Context, labels []string, offset int, limit int) (res responses.MultiDeviceProfilesResponse, edgexError errors.EdgeX) {
+	u, err := url.Parse(client.baseUrl)
+	if err != nil {
+		return res, errors.NewCommonEdgeX(errors.KindClientError, "fail to parse baseUrl", err)
+	}
+	u.Path = path.Join(u.Path, v2.ApiAllDeviceProfileRoute)
+	q := u.Query()
+	if len(labels) > 0 {
+		q.Set(v2.Labels, strings.Join(labels, v2.CommaSeparator))
+	}
+	q.Set(v2.Offset, strconv.Itoa(offset))
+	q.Set(v2.Limit, strconv.Itoa(limit))
+	u.RawQuery = q.Encode()
+	err = utils.GetRequest(ctx, &res, u.String())
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
+}
+
+func (client *DeviceProfileClient) DeviceProfilesByModel(ctx context.Context, model string, offset int, limit int) (res responses.MultiDeviceProfilesResponse, edgexError errors.EdgeX) {
+	u, err := url.Parse(client.baseUrl)
+	if err != nil {
+		return res, errors.NewCommonEdgeX(errors.KindClientError, "fail to parse baseUrl", err)
+	}
+	u.Path = path.Join(u.Path, v2.ApiDeviceProfileRoute, v2.Model, url.QueryEscape(model))
+	q := u.Query()
+	q.Set(v2.Offset, strconv.Itoa(offset))
+	q.Set(v2.Limit, strconv.Itoa(limit))
+	u.RawQuery = q.Encode()
+	err = utils.GetRequest(ctx, &res, u.String())
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
+}
+
+func (client *DeviceProfileClient) DeviceProfilesByManufacturer(ctx context.Context, manufacturer string, offset int, limit int) (res responses.MultiDeviceProfilesResponse, edgexError errors.EdgeX) {
+	u, err := url.Parse(client.baseUrl)
+	if err != nil {
+		return res, errors.NewCommonEdgeX(errors.KindClientError, "fail to parse baseUrl", err)
+	}
+	u.Path = path.Join(u.Path, v2.ApiDeviceProfileRoute, v2.Manufacturer, url.QueryEscape(manufacturer))
+	q := u.Query()
+	q.Set(v2.Offset, strconv.Itoa(offset))
+	q.Set(v2.Limit, strconv.Itoa(limit))
+	u.RawQuery = q.Encode()
+	err = utils.GetRequest(ctx, &res, u.String())
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
+}
+
+func (client *DeviceProfileClient) DeviceProfilesByManufacturerAndModel(ctx context.Context, manufacturer string, model string, offset int, limit int) (res responses.MultiDeviceProfilesResponse, edgexError errors.EdgeX) {
+	u, err := url.Parse(client.baseUrl)
+	if err != nil {
+		return res, errors.NewCommonEdgeX(errors.KindClientError, "fail to parse baseUrl", err)
+	}
+	u.Path = path.Join(u.Path, v2.ApiDeviceProfileRoute, v2.Manufacturer, url.QueryEscape(manufacturer), v2.Model, url.QueryEscape(model))
+	q := u.Query()
+	q.Set(v2.Offset, strconv.Itoa(offset))
+	q.Set(v2.Limit, strconv.Itoa(limit))
+	u.RawQuery = q.Encode()
+	err = utils.GetRequest(ctx, &res, u.String())
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
```

Comment:
This code is repeated (except for the parameters to Join) in most of these functions, . Please refactor into helper function that takes var args for the Join parameters.

Reply:
Refactored the GetRequest function to accept the request path and params.